src/NamingService.ts

```diff
@@ -3,6 +3,7 @@ import { SourceDefinition } from './types';
 export default abstract class NamingService {
   abstract isSupportedDomain(domain: string): boolean;
   abstract isSupportedNetwork(): boolean;
+  abstract resolution(domain: string): Promise<Object | null>;
```

Comment:
IMO `resolution` should be specific method to `Zns`. We are not sure if other naming service would support that.
In case they do, for example, if Ens would support IPFS, we need to implement `Namicorn#ipfs` method. I wouldn't do it now because the format of the method may change and we need to be sure it is stable.

 Lets make `resolution` to return a plain resolution object from `getResolverRecordStructure`.